tests/functional/aws-node-sdk/test/object/deleteObject.js

```diff
@@ -13,70 +20,225 @@ describe('DELETE object', () => {
+                .then(() => {
+                    process.stdout.write('initiating multipart upload\n');
+                    return s3.createMultipartUploadPromise({ Bucket: bucketName,
+                        Key: objectName });
+                })
+                .then(res => {
+                    process.stdout.write('uploading parts\n');
+                    uploadId = res.UploadId;
+                    const uploads = [];
+                    for (let i = 1; i <= 3; i++) {
+                        uploads.push(
+                            s3.uploadPartPromise({ Bucket: bucketName,
+                                Key: objectName, PartNumber: i, Body: testfile,
+                                UploadId: uploadId })
+                        );
+                    }
+                    return Promise.all(uploads);
+                })
+                .catch(err => {
+                    process.stdout.write(`Error with uploadPart ${err}\n`);
+                    throw err;
+                })
+                .then(res => {
+                    process.stdout.write('about to complete multipart ' +
+                        'upload\n');
+                    return s3.completeMultipartUploadPromise({
+                        Bucket: bucketName,
+                        Key: objectName,
+                        UploadId: uploadId,
+                        MultipartUpload: {
+                            Parts: [
+                                { ETag: res[0].ETag, PartNumber: 1 },
+                                { ETag: res[1].ETag, PartNumber: 2 },
+                                { ETag: res[2].ETag, PartNumber: 3 },
+                            ],
+                        },
+                    });
+                })
+                .catch(err => {
+                    process.stdout.write('completeMultipartUpload error: ' +
+                        `${err}\n`);
+                    throw err;
+                });
+            });
+
+            after(() => {
+                process.stdout.write('Emptying bucket\n');
+                return bucketUtil.empty(bucketName)
+                .then(() => {
+                    process.stdout.write('Deleting bucket\n');
+                    return bucketUtil.deleteOne(bucketName);
+                })
+                .catch(err => {
+                    process.stdout.write('Error in after\n');
+                    throw err;
+                });
+            });
+
+            it('should delete a object uploaded in parts successfully',
+            done => {
+                s3.deleteObject({ Bucket: bucketName, Key: objectName },
+                err => {
+                    assert.strictEqual(err, null,
+                        `Expected success, got error ${JSON.stringify(err)}`);
+                    done();
+                });
+            });
+        });
+
+        describe('with object lock', () => {
+            let versionIdOne = '';
+            let versionIdTwo = '';
```

Comment:
I don't think these need to be initialized to empty string, undefined should be fine.
```suggestion
            let versionIdOne;
            let versionIdTwo;
```